spec/std/isa/csr/scountinhibit.yaml

```diff
@@ -0,0 +1,704 @@
+        return csr_value.HPM27
+      }
+  HPM28:
+    location: 28
+    description: |
+      [when="COUNTINHIBIT_EN[28] == true"]
+      When bit is set `hpmcounter28.COUNT` does not increment; when 0, it increments normally.
+
+      [when="COUNTINHIBIT_EN[28] == false"]
+      Since hpmcounter28 is not implemented, this field is read-only zero.
+    type(): |
+      return COUNTINHIBIT_EN[28] ? CsrFieldType::RW : CsrFieldType::RO;
+    reset_value(): |
+      return COUNTINHIBIT_EN[28] ? UNDEFINED_LEGAL : 0;
+    sw_write(csr_value): |
+      if (mode() == PrivilegeMode::M && CSR[menvcfg].CDE == 0) {
+        raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
+      } else if ((mode() == PrivilegeMode::VS && CSR[menvcfg].CDE == 1'b1) ||
+                 (mode() == PrivilegeMode::VU && CSR[menvcfg].CDE == 1'b1)) {
+        raise(ExceptionCode::VirtualInstruction, mode(), $encoding);
+      } else {
+        CSR[mcountinhibit].HPM28 = csr_value.HPM28;
+        return csr_value.HPM28
+      }
+  HPM29:
+    location: 29
+    description: |
+      [when="COUNTINHIBIT_EN[29] == true"]
+      When bit is set `hpmcounter29.COUNT` does not increment; when 0, it increments normally.
+
+      [when="COUNTINHIBIT_EN[29] == false"]
+      Since hpmcounter29 is not implemented, this field is read-only zero.
+    type(): |
+      return COUNTINHIBIT_EN[29] ? CsrFieldType::RW : CsrFieldType::RO;
+    reset_value(): |
+      return COUNTINHIBIT_EN[29] ? UNDEFINED_LEGAL : 0;
+    sw_write(csr_value): |
+      if (mode() == PrivilegeMode::M && CSR[menvcfg].CDE == 0) {
+        raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
+      } else if ((mode() == PrivilegeMode::VS && CSR[menvcfg].CDE == 1'b1) ||
+                 (mode() == PrivilegeMode::VU && CSR[menvcfg].CDE == 1'b1)) {
+        raise(ExceptionCode::VirtualInstruction, mode(), $encoding);
+      } else {
+        CSR[mcountinhibit].HPM29 = csr_value.HPM29;
+        return csr_value.HPM29
+      }
+  HPM30:
+    location: 30
+    description: |
+      [when="COUNTINHIBIT_EN[30] == true"]
+      When bit is set `hpmcounter30.COUNT` does not increment; when 0, it increments normally.
+
+      [when="COUNTINHIBIT_EN[30] == false"]
+      Since hpmcounter30 is not implemented, this field is read-only zero.
+    type(): |
+      return COUNTINHIBIT_EN[30] ? CsrFieldType::RW : CsrFieldType::RO;
+    reset_value(): |
+      return COUNTINHIBIT_EN[30] ? UNDEFINED_LEGAL : 0;
+    sw_write(csr_value): |
+      if (mode() == PrivilegeMode::M && CSR[menvcfg].CDE == 0) {
+        raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
+      } else if ((mode() == PrivilegeMode::VS && CSR[menvcfg].CDE == 1'b1) ||
+                 (mode() == PrivilegeMode::VU && CSR[menvcfg].CDE == 1'b1)) {
+        raise(ExceptionCode::VirtualInstruction, mode(), $encoding);
+      } else {
+        CSR[mcountinhibit].HPM30 = csr_value.HPM30;
+        return csr_value.30
+      }
+  HPM31:
```

Comment:
Given all the repetition, it might be nice to use the "layout" functionality used by other CSRs like `spec/std/isa/csr/I/mcounteren.layout`, which is very similar in that it also has "HPM<n>" fields.

Reply:
If done like the others, there should be a "WARNING" at the top of the generated YAML file (inserted by `insert_warning`), but I don't see that. Did you add the generated YAML file here?